db-service/lib/cqn2sql.js

```diff
@@ -805,9 +805,8 @@ class CQN2SQLRenderer {
    */
   ref({ ref }) {
     switch (ref[0]) {
-      case '$now': return this.func({ func: 'session_context', args: [{ val: '$now', param: false }] })
-      case '$user':
-      case '$user.id': return this.func({ func: 'session_context', args: [{ val: '$user.id', param: false }] })
+      case '$now':  return this.func({ func: 'session_context', args: [{ val: '$now', param: false }] }) // REVISIT: why do we need param: false here?
+      case '$user': return this.func({ func: 'session_context', args: [{ val: '$user.'+ref[1]||'id', param: false }] }) // REVISIT: same here?
```

Comment:
@BobdenOs can you answer the REVISIT questions? → Why do we need `param: false` in here?
If there is a reason, we should know/document it, like in a comment somewhere.
If not, we should remove it.

Reply:
It is required to reduce the amount of `values` used inside queries. As they cannot change it is best to have them embedded into the SQL. That way the optimizer doesn't need to consider all possible values. Additionally it is not required to send the value every execution.

```suggestion
      // param: false is used to ensure the value is embedded into the SQL as it will not change
      case '$now':  return this.func({ func: 'session_context', args: [{ val: '$now', param: false }] })
      case '$user': return this.func({ func: 'session_context', args: [{ val: '$user.'+ref[1]||'id', param: false }] })
```